runtime/src/main/resources/reference.conf

```diff
@@ -54,9 +58,9 @@ akka.grpc.client."*" {
   # interpreted as retrying 'indefinitely'.
   connection-attempts = 20
 
-  # Service discovery mechamism to use. The default is to use a static host
-  # and port that will be resolved via DNS.
-  # Any of the mechanisms described in https://doc.akka.io/docs/akka-management/current/discovery/index.html can be used
-  # including Kubernetes, Consul, AWS API
+  # Request that the client try to connect the service immediately when the client is created
+  # rather than on the first request. Only supported for the Netty client backend, the akka-http client backend
+  # is always eager.
+  eager-connection = off
```

Comment:
Shall we add something to the client reference docs about this? Seems like something you want to enable most of the time.

Reply:
I'm not entirely sure it is something you want enabled most of the time, it since it would also means you might create a bunch of connections for clients that don't see any traffic so wastes resource usage (at least I'm guessing that is why it is lazy by default). For the record clients will also idle timeout after a default 5 minutes without any requests and switch to an "IDLE" state and only reconnect when a new request is made.

Reply:
Mentioning in docs seems good though.

Reply:
ok, yes we should not change the default

Reply:
Added in [6222128](https://github.com/akka/akka-grpc/pull/1931/commits/6222128c97930bf54839cdc1f28374d00a67da09)